xdsl/interactive/app.py

```diff
@@ -0,0 +1,140 @@
+"""
+This app allows you to paste xDSL IR into the Input TextArea, select a pass (or multiple passes) to be applied on the input IR
```

Comment:
This is a great description of the current behaviour in the other PR, but may quickly become outdated as the code is changed. I would recommend describing the purpose/overall description of the contents of this file, rather than a description of the behaviour. Maybe something along these lines:

```
An interactive command-line tool to explore compilation pipeline construction.

Execute `xdsl-gui` in your terminal to run it.

Run `terminal -m xdsl.interactive.app:InputApp --def` to run in development mode. Please be sure to install `textual-dev` to run this command.
```

Another thing I forgot to suggest is to install this as a script along with `xdsl-opt` and `xdsl-run`, which we may discuss in this PR or an upcoming one. Maybe best left for later.

Reply:
I agree, the top-level command is best left for later, but it would be good for this documentation to already be a little more instructive to people who come across this file without seeing this PR.